monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -4249,10 +4249,18 @@ object Task extends TaskInstancesLevel1 {
     *        case you want the [[monix.execution.misc.Local Local]]
     *        variables to be propagated on async boundaries.
     *        Defaults to `false`.
+    * 
+    * @param isolateLocalContextOnRunToFuture should be set to
+    *        `true` if when using `localContextPropagation` you
+    *        want the [[monix.execution.misc.Local Local]] to be
+    *        isolated when running the task otherwise the
+    *        [[monix.execution.misc.Local Local]] will be visible after
+    *        running to a Future. Defaults to `true`.
     */
   final case class Options(
```

Comment:
If you follow Oleg's PR: https://github.com/monix/monix/pull/1044/files
then we should be able to preserve binary compatibility for at least Scala 2.12 and 2.13.

You can check if it passes with something like:
```scala
++ 2.12.10
evalJVM/mimaReportBinaryIssues
```

I feel like we can get away with only the following in companion object (+ similar changes in case class):
```scala
  object Options extends AbstractFunction2[Boolean, Boolean, Options] {
    def apply(autoCancelableRunLoops: Boolean, localContextPropagation: Boolean) =
      new Options(autoCancelableRunLoops, localContextPropagation)
  }
```

But I'm not 100% sure yet.

Reply:
So I tried replicating @oleg-py PR as much as possible however I am still getting

```
[error] monix-eval: Failed binary compatibility check against io.monix:monix-eval_2.12:3.0.0! Found 1 potential problems
[error]  * the type hierarchy of object monix.eval.Task#Options is different in current version. Missing types {scala.runtime.AbstractFunction2}
[error]    filter with: ProblemFilters.exclude[MissingTypesProblem]("monix.eval.Task$Options$")
[error] stack trace is suppressed; run last evalJVM / mimaReportBinaryIssues for the full output
[error] (evalJVM / mimaReportBinaryIssues) Failed binary compatibility check against io.monix:monix-eval_2.12:3.0.0! Found 1 potential problems
[error] Total time: 2 s, completed Mar 28, 2020 12:23:41 PM
```

Reply:
Yes, I feel like we only need `def apply` there, at least according to mima and our tests :D